encoding/ssh/filexfer/filexfer.go

```diff
@@ -0,0 +1,12 @@
+package filexfer
+
+// Packet defines the behavior of an SFTP packet.
+type Packet interface {
+	MarshalPacket(reqid uint32) (header, payload []byte, err error)
+	UnmarshalPacketBody(buf *Buffer) error
+}
+
+// ComposePacket converts returns from MarshalPacket into an equivalent call to MarshalBinary.
+func ComposePacket(header, payload []byte, err error) ([]byte, error) {
```

Comment:
When would this function be used? It looks like it could be private, at least for now, but I haven't considered any server code.

Reply:
The situation is for other versions of the library being implemented in a sort of “overlay” format. So `encoding/ssh/filexfer/v4` could make use of this function as well to quickly convert its `MarshalPacket` into a `MarshalBinary` for its own `SSH_FXP_NAME` type (which is binary incompatible with v3’s version), or like `encoding/ssh/filexfer/v6` with the `SSH_FXP_LINK` and `SSH_FXP_BLOCK` and `SSH_FXP_UNBLOCK`. 

This library is intended to be flexible enough to implement all versions of the `filexfer` protocol, not just strictly the v3. Though, v3 is the default and main protocol implementation.